netpyne/sim/load.py

```diff
@@ -451,6 +451,57 @@ def loadAll(filename, data=None, instantiate=True, createNEURONObj=True):
     loadNet(filename, data=data, instantiate=instantiate, compactConnFormat=connFormat)
     loadSimData(filename, data=data)
 
+def loadFromIndex(index, method='python'):
+
+    import __main__
+    import json, os, importlib, types
+    from .. import sim
+
+    print(f'Loading index file {index} ... ')
+    with open(index, 'r') as fileObj:
+        indexData = json.load(fileObj)
+
+        # compile .mod files
+        modFolder = indexData['mod_folder']
+        os.system('nrnivmodl ' + modFolder)
+
+        if method == 'python':
+            simConfigFile = indexData['simConfig_python']
+            print(f'\n    Loading simConfig: {simConfigFile} ... ')
+            loader = importlib.machinery.SourceFileLoader(os.path.basename(simConfigFile).split('.')[0], simConfigFile)
+            cfgModule = types.ModuleType(loader.name)
+            loader.exec_module(cfgModule)
+            cfg = cfgModule.cfg
+            __main__.cfg = cfg
```

Comment:
Hmm, I think I did this because inside `netPrams.py` there was a reference to cfg module, and I was just trying to see it all works. But yeah, looks like a hack for this particular case. Let me check again

Reply:
it turns out indeed, there are quite a lot of places, where `netParams.py` has the following code: `from __main__ import cfg`. So if we don't set cfg to __main__ before loading netParams module, it crashes. This is the case for about 10 of examples. Maybe we should keep this `__main__.cfg = cfg` in `loadFromIndexFile()`, at least flagged by a parameter?

Reply:
ah yeah that makes sense, most of the batch examples require this. ok so let's leave it for now, if we need to make it optional with a flag we can add that later.